themes/default/content/blog/serverless-architecture-design/index.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "Modern Serverless Architecture Design"
+date: 2023-11-03
+draft: false
+meta_desc: "Learn about the concept of serverless architecture and the different considerations to make when building out your serverless application."
+meta_image: meta.png
+authors:
+    - adora-nwodo
+tags:
+    - serverless
+    - serverless-functions
+    - azure-functions
+    - system-design
+    - system-architecture
+
+---
+
+Serverless architectures have become popular because they make it easier for software developers to focus on application development without the additional operation overhead. With Pulumi, you can deploy and manage your serverless infrastructure and this article shows you how to do this with Azure.
+
+Before we go into the specifics of serverless architecture design with Pulumi on azure, let's go
+through the basics of serverless - this is important for the rest of the article.
+
+## What is Serverless Architecture
+
+Serverless architecture is a cloud computing model that simplifies the management of infrastructure. This allows software developers to focus on writing code without worrying about server provisioning, scaling or maintenance. The idea behind it is functions as a service (FaaS), in which the environment dynamically scales in response to demand and code is executed in response to events. In the serverless architecture, developers focus on writing and deploying code in the form of serverless functions, which are executed in response to specific events or triggers.
```

Comment:
You mention FaaS here, but can we also make connections/comparisons with things like event-driven architectures? That seems relevant, but perhaps I'm mistaken. I think it would also be helpful to mention specific product offerings here---what are the serverless offerings from AWS? From Azure? From Google Cloud?

Reply:
As for event-driven, you do mention that serverless/FaaS _is_ event-driven. Here, I was thinking of you explaining/mentioning "event-driven architecture (EDA)" since it is a term that lots of folks run into/see. Is it the same as serverless? (Not necessarily. Serverless is EDA, but not all EDA is serverless.) It doesn't need to be a lot, but enough to help folks understand some of the different pieces involved.